src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.h

```diff
@@ -0,0 +1,175 @@
+    As long as there are at least two clusters, the alignment is done as follows:\n
+    Of every pair of clusters, the one with the larger 10/90 percentile retention time range is selected as reference for the align() method of @ref OpenMS::MapAlignmentAlgorithmIdentification.
+    align() maps the median retention time of each ID group in the second cluster to the reference retention time of this group.
+    Cubic spline smoothing is used to convert this mapping to a smooth function.
+    Retention times in the second cluster are transformed to the reference scale by applying this function.
+    Additionally, the original retention times are stored in the meta information of each ID group.
+    The reference is combined with the transformed cluster.
+
+    The resulting map is used to extract transformation descriptions for each input map.
+    For each map cubic spline smoothing is used to convert the mapping to a smooth function.
+    Retention times of each map are transformed applying corresponding function.
+
+    @htmlinclude OpenMS_MapAlignmentAlgorithmTreeGuided.parameters
+
+    @ingroup MapAlignment
+
+  */
+  class OPENMS_DLLAPI MapAlignmentAlgorithmTreeGuided :
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+public:
+    /// Default constructor
+    MapAlignmentAlgorithmTreeGuided();
+
+    /// Destructor
+    ~MapAlignmentAlgorithmTreeGuided() override;
+
+    /**
+     * @brief Extract RTs given for individual features of each map, calculate distances for each pair of maps and cluster hierarchical using average linkage.
+     *
+     * @param feature_maps Vector of input maps (FeatureMap) whose distance is to be calculated.
+     * @param tree Vector of BinaryTreeNodes that will be computed
+     * @param maps_ranges Vector to store all sorted RTs of extracted identifications for each map in @p feature_maps; needed to determine the 10/90 percentiles
+    */
+    static void buildTree_(std::vector<FeatureMap>& feature_maps, std::vector<BinaryTreeNode>& tree, std::vector<std::vector<double>>& maps_ranges);
+
+    /**
+     * @brief Align feature maps tree guided using align() of @ref OpenMS::MapAlignmentAlgorithmIdentification and use TreeNode with larger 10/90 percentile range as reference.
+     *
+     * @param tree Vector of BinaryTreeNodes that contains order for alignment.
+     * @param feature_maps_transformed Vector with copied input maps for transformation process.
+     * @param maps_ranges Vector that contains all sorted RTs of extracted identifications for each map; needed to determine the 10/90 percentiles.
+     * @param map_transformed FeatureMap to store all features of combined maps with original and transformed RTs in order of alignment.
+     * @param trafo_order Vector to store indices of maps in order of alignment.
+    */
+    void treeGuidedAlignment_(const std::vector<BinaryTreeNode> &tree, std::vector<FeatureMap> feature_maps_transformed,
```

Comment:
can this be static?

Reply:
Only if the `defaults_` parameters are passed with the method instead.